cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -34,3 +39,154 @@ def test_cluster_single_db(cluster_with_single_db, logger, attributes, cfy):
                      cert_path=mgr2.local_ca)
 
     check_managers(mgr1, mgr2)
+
+
+def test_queue_node_failover(cluster_with_single_db, logger,
+                             module_tmpdir, attributes, ssh_key, cfy):
+    hello_world = prepare_cluster_with_agent(
+        cluster_with_single_db, logger,
+        module_tmpdir, attributes, ssh_key, cfy
+    )
+    broker1, broker2, broker3, db, mgr1, mgr2 = cluster_with_single_db
+    validate_cluster_and_agents(mgr1)
+    agent_broker_ip = verify_agent_broker_connection_and_get_broker_ip(mgr1)
+
+    # stop the rabbitmq service in the agent's broker node
+    agent_broker = None
+    for broker in [broker1, broker2, broker3]:
+        if broker.private_ip_address == agent_broker_ip:
+            agent_broker = broker
+            break
+    with agent_broker.ssh() as broker_ssh:
+        broker_ssh.run('sudo service cloudify-rabbitmq stop')
+
+    # the agent should now pick another broker
+    validate_cluster_and_agents(mgr1, expected_broker_status='Degraded')
+    new_agent_broker_ip = \
+        verify_agent_broker_connection_and_get_broker_ip(mgr1)
+    assert new_agent_broker_ip != agent_broker_ip
+
+    new_agent_broker = None
+    for broker in [broker1, broker2, broker3]:
+        if broker.private_ip_address == new_agent_broker_ip:
+            new_agent_broker = broker
+            break
+
+    # only 2 brokers should now be connected to the manager and agents
+    with new_agent_broker.ssh() as broker_ssh:
+        active_brokers = broker_ssh.run(
+            'sudo -u rabbitmq rabbitmqctl list_connections host'.format(
+                broker.hostname)).split('\r\n')[1:]
+        assert len(set(active_brokers)) == 2
+
+    # bring the stopped broker back up
+    with agent_broker.ssh() as broker_ssh:
+        broker_ssh.run('sudo service cloudify-rabbitmq start')
+
+    # spin up 3 deployments to check whether agents will pick up the new broker
```

Comment:
If this entire section is intended to validate that agents can use all brokers then this is a fragile (the brokers are picked randomly so it's possible that not all of them will be picked) and expensive (lots of extra VMs deployed).

A better way might be to just use the deployment you already have, blocking access to all but one of the brokers, restarting the agent, then confirming it connected, and repeating that for each broker?